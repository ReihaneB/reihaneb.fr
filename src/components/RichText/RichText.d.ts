import type { BlocksContent } from '@strapi/blocks-react-renderer';

export interface RichTextProps {
  content: BlocksContent;
}
