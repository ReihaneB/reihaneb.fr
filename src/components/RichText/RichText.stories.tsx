/* eslint-disable @typescript-eslint/ban-ts-comment */
// TODO: Find a way to test images in the RichText component without typescript errors (see below)
// @ts-nocheck

import type { Meta, StoryObj } from '@storybook/react';

import _1x1 from '@/styles/assets/placeholder/_1x1.png';

import RichText from './RichText';

const meta: Meta<typeof RichText> = {
  title: 'Components/RichText',
  component: RichText,
};

export default meta;
type Story = StoryObj<typeof RichText>;

export const Playground: Story = {
  args: {
    content: [
      {
        type: 'heading',
        level: 1,
        children: [
          {
            text: 'Titre de l\'article',
            type: 'text',
          },
        ],
      },
      {
        type: 'paragraph',
        children: [
          {
            text: 'Consequat sunt elit deserunt anim cupidatat id proident mollit deserunt.',
            type: 'text',
          },
        ],
      },
      {
        type: 'heading',
        level: 2,
        children: [
          {
            text: 'Premier paragraphe',
            type: 'text',
          },
        ],
      },
      {
        type: 'paragraph',
        children: [
          {
            text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus. Suspendisse lectus tortor, dignissim sit amet, adipiscing nec, ultricies sed, dolor. Cras elementum ultrices diam. Maecenas ligula ',
            type: 'text',
          },
          {
            url: 'https://liome.fr',
            type: 'link',
            children: [
              {
                text: 'massa',
                type: 'text',
              },
            ],
          },
          {
            text: ', varius a, semper congue, euismod non, mi.',
            type: 'text',
          },
        ],
      },
      {
        type: 'heading',
        level: 2,
        children: [
          {
            text: 'Image',
            type: 'text',
          },
        ],
      },
      {
        type: 'image',
        image: {
          url: _1x1,
          width: 400,
          height: 400,
          alternativeText: 'A placeholder image',
        },
      },
      {
        type: 'heading',
        level: 2,
        children: [
          {
            text: 'Deuxième paragraphe',
            type: 'text',
          },
        ],
      },
      {
        type: 'paragraph',
        children: [
          {
            text: 'Proin porttitor, orci nec nonummy molestie, enim est eleifend mi, non fermentum diam nisl sit amet erat. Duis semper. Duis arcu massa, scelerisque vitae, consequat in, pretium a, enim. Pellentesque congue. Ut in risus volutpat libero pharetra tempor. Cras vestibulum bibendum augue. Praesent egestas leo in pede. Praesent blandit odio eu enim.',
            type: 'text',
          },
        ],
      },
      {
        type: 'heading',
        level: 2,
        children: [
          {
            text: 'Liste ordonnée',
            type: 'text',
          },
        ],
      },
      {
        type: 'list',
        format: 'ordered',
        children: [
          {
            type: 'list-item',
            children: [
              {
                text: 'Integer nec odio.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Praesent libero.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Sed cursus ante dapibus diam.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Sed nisi.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Nulla quis sem at nibh elementum imperdiet.',
                type: 'text',
              },
            ],
          },
        ],
      },
      {
        type: 'heading',
        level: 2,
        children: [
          {
            text: 'Quote',
            type: 'text',
          },
        ],
      },
      {
        type: 'quote',
        children: [
          {
            text: 'Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.',
            type: 'text',
          },
        ],
      },
      {
        type: 'heading',
        level: 2,
        children: [
          {
            text: 'Liste non ordonnée',
            type: 'text',
          },
        ],
      },
      {
        type: 'list',
        format: 'unordered',
        children: [
          {
            type: 'list-item',
            children: [
              {
                text: 'Curabitur sodales ligula in libero.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Sed dignissim lacinia nunc.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Curabitur tortor.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Pellentesque nibh.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Aenean quam.',
                type: 'text',
              },
            ],
          },
        ],
      },
      {
        type: 'heading',
        level: 2,
        children: [
          {
            text: 'Troisième paragraphe',
            type: 'text',
          },
        ],
      },
      {
        type: 'paragraph',
        children: [
          {
            text: 'In scelerisque sem at dolor. Maecenas mattis. Sed convallis tristique sem. Proin ut ligula vel nunc egestas porttitor. Morbi lectus risus, iaculis vel, suscipit quis, luctus non, massa. Fusce ac turpis quis ligula lacinia aliquet.',
            type: 'text',
          },
        ],
      },
      {
        type: 'heading',
        level: 2,
        children: [
          {
            text: 'Code',
            type: 'text',
          },
        ],
      },
      {
        type: 'code',
        children: [
          {
            text: '```python\ndef saluer(nom):\n print(f"Bonjour, {nom}!")\n saluer("Lorem Ipsum")',
            type: 'text',
          },
        ],
      },
      {
        type: 'paragraph',
        children: [
          {
            text: '',
            type: 'text',
          },
        ],
      },
    ],
  },
};

export const Paragraph: Story = {
  args: {
    content: [
      {
        type: 'paragraph',
        children: [
          {
            text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus. Suspendisse lectus tortor, dignissim sit amet, adipiscing nec, ultricies sed, dolor. Cras elementum ultrices diam. Maecenas ligula ',
            type: 'text',
          },
          {
            url: 'https://liome.fr',
            type: 'link',
            children: [
              {
                text: 'massa',
                type: 'text',
              },
            ],
          },
          {
            text: ', varius a, semper congue, euismod non, mi.',
            type: 'text',
          },
        ],
      },
    ],
  },
  name: 'Paragraph',
};

export const Heading: Story = {
  args: {
    content: [
      {
        type: 'heading',
        level: 1,
        children: [
          {
            text: 'Titre de l\'article',
            type: 'text',
          },
        ],
      },
    ],
  },
  name: 'Heading',
};

// An error occurs when trying to display an image in the RichText component
// because the image object is not correctly typed in the BlocksContent.d.ts file
// to test with a StaticImageData object
export const Image: Story = {
  args: {
    content: [
      {
        type: 'image',
        image: {
          url: _1x1,
          alternativeText: 'A placeholder image',
        },
      },
    ],
  },
  name: 'Image',
};

export const OrderedList: Story = {
  args: {
    content: [
      {
        type: 'list',
        format: 'ordered',
        children: [
          {
            type: 'list-item',
            children: [
              {
                text: 'Integer nec odio.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Praesent libero.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Sed cursus ante dapibus diam.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Sed nisi.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Nulla quis sem at nibh elementum imperdiet.',
                type: 'text',
              },
            ],
          },
        ],
      },
    ],
  },
  name: 'Ordered List',
};

export const UnorderedList: Story = {
  args: {
    content: [
      {
        type: 'list',
        format: 'unordered',
        children: [
          {
            type: 'list-item',
            children: [
              {
                text: 'Curabitur sodales ligula in libero.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Sed dignissim lacinia nunc.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Curabitur tortor.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Pellentesque nibh.',
                type: 'text',
              },
            ],
          },
          {
            type: 'list-item',
            children: [
              {
                text: 'Aenean quam.',
                type: 'text',
              },
            ],
          },
        ],
      },
    ],
  },
  name: 'Unordered List',
};

export const Quote: Story = {
  args: {
    content: [
      {
        type: 'quote',
        children: [
          {
            text: 'Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.',
            type: 'text',
          },
        ],
      },
    ],
  },
  name: 'Quote',
};

export const Code: Story = {
  args: {
    content: [
      {
        type: 'code',
        children: [
          {
            text: '```python\ndef saluer(nom):\n print(f"Bonjour, {nom}!")\n saluer("Lorem Ipsum")',
            type: 'text',
          },
        ],
      },
    ],
  },
  name: 'Code',
};
