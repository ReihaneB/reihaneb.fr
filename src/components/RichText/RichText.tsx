import type React from 'react';
import { BlocksRenderer } from '@strapi/blocks-react-renderer';

import Link from '../Link/Link';
import Image from '../Image/Image';
import SyntaxHighlighter from '../SyntaxHighlighter/SyntaxHighlighter';
import Blockquote from '../Blockquote/Blockquote';
import List from '../List/List';

import styles from './RichText.module.css';
import { RichTextProps } from './RichText.d';

function extractLanguage(codeBlock: string) {
  const match = codeBlock.match(/```(\w+)/);
  return match ? match[1] : null;
}

function RichText({
  content,
  ...rest
}: RichTextProps) {
  return (
    <div className={styles.root} {...rest}>
      <BlocksRenderer
        content={content}
        blocks={{
          image: ({ image }) => (
            <Image
              src={image.url}
              alt={image.alternativeText || ''}
              priority
            />
          ),
          link: ({ children, url }) => (
            <Link size="small" to={url}>
              {children}
            </Link>
          ),
          list: ({ children, format }) => (
            <List type={format}>
              {Array.isArray(children) ? children : [children]}
            </List>
          ),
          quote: ({ children }) => (
            <Blockquote>
              {Array.isArray(children) ?
                children.map((child: React.ReactElement) => child.props.text).join('\n') :
                ''}
            </Blockquote>
          ),
          code: ({ children }) => {
            if (Array.isArray(children)) {
              const combinedText = children
                .map((child: React.ReactElement) => child.props.text.replace(/^```(\w+)\n/, ''))
                .join('\n');

              const lang = extractLanguage(children.map((child: React.ReactElement) => child.props.text).join('\n')) || undefined;

              return (
                <SyntaxHighlighter language={lang}>
                  {combinedText}
                </SyntaxHighlighter>
              );
            }
            return null;
          },
        }}
      />
    </div>
  );
}

export default RichText;
