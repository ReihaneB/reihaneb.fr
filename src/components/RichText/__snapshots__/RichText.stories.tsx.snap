// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/RichText Code smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-RichText-RichText-module__root--R2tKv">
    <div class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__root--Cc7Sr">
      <p class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__language--dWL63">
        python
      </p>
      <pre style="color: rgb(17, 27, 39); background: transparent; font-family: Consolas, Monaco, &quot;Andale Mono&quot;, &quot;Ubuntu Mono&quot;, monospace; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; overflow-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: 0px; overflow: auto;">
        <code style="background-color: transparent; font-family: &quot;Fira Code&quot;, &quot;Fira Mono&quot;, Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace; padding: 0px; white-space: pre;">
          <span class="token"
                style="color: rgb(160, 73, 0);"
          >
            def
          </span>
          <span>
          </span>
          <span class="token"
                style="color: rgb(124, 0, 170);"
          >
            saluer
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            (
          </span>
          <span>
            nom
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            )
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            :
          </span>
          <span>
          </span>
          <span>
          </span>
          <span class="token"
                style="color: rgb(160, 73, 0);"
          >
            print
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            (
          </span>
          <span class="token string-interpolation"
                style="color: rgb(17, 107, 0);"
          >
            f"Bonjour,
          </span>
          <span class="token string-interpolation interpolation"
                style="color: rgb(17, 27, 39);"
          >
            {
          </span>
          <span class="token string-interpolation interpolation">
            nom
          </span>
          <span class="token string-interpolation interpolation"
                style="color: rgb(17, 27, 39);"
          >
            }
          </span>
          <span class="token string-interpolation"
                style="color: rgb(17, 107, 0);"
          >
            !"
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            )
          </span>
          <span>
          </span>
          <span>
            saluer
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            (
          </span>
          <span class="token"
                style="color: rgb(17, 107, 0);"
          >
            "Lorem Ipsum"
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            )
          </span>
        </code>
      </pre>
    </div>
  </div>
</main>
`;

exports[`Components/RichText Heading smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-RichText-RichText-module__root--R2tKv">
    <h1>
      Titre de l'article
    </h1>
  </div>
</main>
`;

exports[`Components/RichText Image smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-RichText-RichText-module__root--R2tKv">
    <img alt="A placeholder image"
         fetchpriority="high"
         width="1080"
         height="1080"
         decoding="async"
         data-nimg="1"
         class="src-components-Image-Image-module__image--gXjYq"
         srcset="/static/media/_1x1.ea48a62d.png?w=1080&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=3840&amp;q=75 2x"
         src="/static/media/_1x1.ea48a62d.png?w=3840&amp;q=75"
         style="color: transparent;"
    >
  </div>
</main>
`;

exports[`Components/RichText OrderedList smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-RichText-RichText-module__root--R2tKv">
    <ol class="src-components-List-List-module__root--Z5ozz">
      <li>
        Integer nec odio.
      </li>
      <li>
        Praesent libero.
      </li>
      <li>
        Sed cursus ante dapibus diam.
      </li>
      <li>
        Sed nisi.
      </li>
      <li>
        Nulla quis sem at nibh elementum imperdiet.
      </li>
    </ol>
  </div>
</main>
`;

exports[`Components/RichText Paragraph smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-RichText-RichText-module__root--R2tKv">
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus. Suspendisse lectus tortor, dignissim sit amet, adipiscing nec, ultricies sed, dolor. Cras elementum ultrices diam. Maecenas ligula
      <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__small--nBuK2"
         data-testid="link-component"
         target="_blank"
         rel="noreferrer"
         href="https://liome.fr"
      >
        massa
      </a>
      , varius a, semper congue, euismod non, mi.
    </p>
  </div>
</main>
`;

exports[`Components/RichText Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-RichText-RichText-module__root--R2tKv">
    <h1>
      Titre de l'article
    </h1>
    <p>
      Consequat sunt elit deserunt anim cupidatat id proident mollit deserunt.
    </p>
    <h2>
      Premier paragraphe
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus. Suspendisse lectus tortor, dignissim sit amet, adipiscing nec, ultricies sed, dolor. Cras elementum ultrices diam. Maecenas ligula
      <a class="src-components-Link-Link-module__root--ggyqZ src-components-Link-Link-module__primary--hzP7j src-components-Link-Link-module__small--nBuK2"
         data-testid="link-component"
         target="_blank"
         rel="noreferrer"
         href="https://liome.fr"
      >
        massa
      </a>
      , varius a, semper congue, euismod non, mi.
    </p>
    <h2>
      Image
    </h2>
    <img alt="A placeholder image"
         fetchpriority="high"
         width="400"
         height="400"
         decoding="async"
         data-nimg="1"
         class="src-components-Image-Image-module__image--gXjYq"
         srcset="/static/media/_1x1.ea48a62d.png?w=640&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=828&amp;q=75 2x"
         src="/static/media/_1x1.ea48a62d.png?w=828&amp;q=75"
         style="color: transparent;"
    >
    <h2>
      Deuxième paragraphe
    </h2>
    <p>
      Proin porttitor, orci nec nonummy molestie, enim est eleifend mi, non fermentum diam nisl sit amet erat. Duis semper. Duis arcu massa, scelerisque vitae, consequat in, pretium a, enim. Pellentesque congue. Ut in risus volutpat libero pharetra tempor. Cras vestibulum bibendum augue. Praesent egestas leo in pede. Praesent blandit odio eu enim.
    </p>
    <h2>
      Liste ordonnée
    </h2>
    <ol class="src-components-List-List-module__root--Z5ozz">
      <li>
        Integer nec odio.
      </li>
      <li>
        Praesent libero.
      </li>
      <li>
        Sed cursus ante dapibus diam.
      </li>
      <li>
        Sed nisi.
      </li>
      <li>
        Nulla quis sem at nibh elementum imperdiet.
      </li>
    </ol>
    <h2>
      Quote
    </h2>
    <blockquote class="src-components-Blockquote-Blockquote-module__root--LtAGd">
      <p class="src-components-Blockquote-Blockquote-module__text--H_BLP">
        Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
      </p>
    </blockquote>
    <h2>
      Liste non ordonnée
    </h2>
    <ul class="src-components-List-List-module__root--Z5ozz">
      <li>
        Curabitur sodales ligula in libero.
      </li>
      <li>
        Sed dignissim lacinia nunc.
      </li>
      <li>
        Curabitur tortor.
      </li>
      <li>
        Pellentesque nibh.
      </li>
      <li>
        Aenean quam.
      </li>
    </ul>
    <h2>
      Troisième paragraphe
    </h2>
    <p>
      In scelerisque sem at dolor. Maecenas mattis. Sed convallis tristique sem. Proin ut ligula vel nunc egestas porttitor. Morbi lectus risus, iaculis vel, suscipit quis, luctus non, massa. Fusce ac turpis quis ligula lacinia aliquet.
    </p>
    <h2>
      Code
    </h2>
    <div class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__root--Cc7Sr">
      <p class="src-components-SyntaxHighlighter-SyntaxHighlighter-module__language--dWL63">
        python
      </p>
      <pre style="color: rgb(17, 27, 39); background: transparent; font-family: Consolas, Monaco, &quot;Andale Mono&quot;, &quot;Ubuntu Mono&quot;, monospace; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; overflow-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: 0px; overflow: auto;">
        <code style="background-color: transparent; font-family: &quot;Fira Code&quot;, &quot;Fira Mono&quot;, Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace; padding: 0px; white-space: pre;">
          <span class="token"
                style="color: rgb(160, 73, 0);"
          >
            def
          </span>
          <span>
          </span>
          <span class="token"
                style="color: rgb(124, 0, 170);"
          >
            saluer
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            (
          </span>
          <span>
            nom
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            )
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            :
          </span>
          <span>
          </span>
          <span>
          </span>
          <span class="token"
                style="color: rgb(160, 73, 0);"
          >
            print
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            (
          </span>
          <span class="token string-interpolation"
                style="color: rgb(17, 107, 0);"
          >
            f"Bonjour,
          </span>
          <span class="token string-interpolation interpolation"
                style="color: rgb(17, 27, 39);"
          >
            {
          </span>
          <span class="token string-interpolation interpolation">
            nom
          </span>
          <span class="token string-interpolation interpolation"
                style="color: rgb(17, 27, 39);"
          >
            }
          </span>
          <span class="token string-interpolation"
                style="color: rgb(17, 107, 0);"
          >
            !"
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            )
          </span>
          <span>
          </span>
          <span>
            saluer
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            (
          </span>
          <span class="token"
                style="color: rgb(17, 107, 0);"
          >
            "Lorem Ipsum"
          </span>
          <span class="token"
                style="color: rgb(17, 27, 39);"
          >
            )
          </span>
        </code>
      </pre>
    </div>
    <br>
  </div>
</main>
`;

exports[`Components/RichText Quote smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-RichText-RichText-module__root--R2tKv">
    <blockquote class="src-components-Blockquote-Blockquote-module__root--LtAGd">
      <p class="src-components-Blockquote-Blockquote-module__text--H_BLP">
        Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
      </p>
    </blockquote>
  </div>
</main>
`;

exports[`Components/RichText UnorderedList smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <div class="src-components-RichText-RichText-module__root--R2tKv">
    <ul class="src-components-List-List-module__root--Z5ozz">
      <li>
        Curabitur sodales ligula in libero.
      </li>
      <li>
        Sed dignissim lacinia nunc.
      </li>
      <li>
        Curabitur tortor.
      </li>
      <li>
        Pellentesque nibh.
      </li>
      <li>
        Aenean quam.
      </li>
    </ul>
  </div>
</main>
`;
